core/src/main/java/com/sequenceiq/cloudbreak/service/ClusterCreationSetupService.java

```diff
@@ -178,54 +183,47 @@ public Cluster prepare(ClusterV4Request request, Stack stack, User user, Workspa
     public Cluster prepare(ClusterV4Request request, Stack stack, ClusterDefinition clusterDefinition, User user, Workspace workspace) throws IOException,
             CloudbreakImageNotFoundException, TransactionExecutionException {
         String stackName = stack.getName();
-        Cluster cluster = stack.getCluster();
+        Cluster clusterStub = stack.getCluster();
         stack.setCluster(null);
 
-        decorateHostGroupWithConstraint(stack, cluster);
-
-        long start = System.currentTimeMillis();
+        decorateHostGroupWithConstraint(stack, clusterStub);
 
         if (request.getCloudStorage() != null) {
-            FileSystem fs = fileSystemConfigService.create(converterUtil.convert(request.getCloudStorage(), FileSystem.class), stack.getWorkspace(),
-                    stack.getCreator());
-            LOGGER.debug("File system saving took {} ms for stack {}", System.currentTimeMillis() - start, stackName);
+            FileSystem fs = b(() -> fileSystemConfigService.create(converterUtil.convert(request.getCloudStorage(), FileSystem.class), stack.getWorkspace(),
```

Comment:
This is a generic comment. Instead of having the b() in the code, wouldn't it be simpler to read the code if methods are annotated with a Mesure annotation? I understood that we might lose the granularity, but that can be solved by refactoring the code.